src/internal/group_api/mod.rs

```diff
@@ -297,47 +359,75 @@ pub(crate) fn get_group_keys<'a>(
 /// `user_master_pub_key` - public key of the user creating this group.
 /// `group_id` - unique id for the group within the segment.
 /// `name` - name for the group. Does not need to be unique.
-/// `add_as_member` - if true the user represented by the current DeviceContext will also be added to the group's membership.
-///     If false, the user will not be an member (but will still be an admin)
+/// `members` - list of user ids to add as members of the group. This list takes priority over `add_as_member`,
+///     so the calling user will be added as a member if their id is in this list even if `add_as_member` is false.
 /// `needs_rotation` - true if the group private key should be rotated by an admin, else false
 pub fn group_create<'a, CR: rand::CryptoRng + rand::RngCore>(
     recrypt: &'a Recrypt<Sha256, Ed25519, RandomBytes<CR>>,
     auth: &'a RequestAuth,
     user_master_pub_key: &'a PublicKey,
     group_id: Option<GroupId>,
     name: Option<GroupName>,
-    add_as_member: bool,
+    members: &'a Vec<UserId>,
     needs_rotation: bool,
-) -> impl Future<Item = GroupMetaResult, Error = IronOxideErr> + 'a {
-    transform::gen_group_keys(recrypt)
-        .and_then(move |(plaintext, group_priv_key, group_pub_key)| {
-            // encrypt the group secret to the current user as they will be the group admin
-            Ok({
-                let encrypted_group_key = recrypt.encrypt(
-                    &plaintext,
-                    &user_master_pub_key.into(),
-                    &auth.signing_private_key().into(),
-                )?;
-
-                // compute a transform key if we are adding the caller as a group member as well
-                let transform_key: Option<crate::internal::TransformKey> = if add_as_member {
-                    Some(
-                        recrypt
-                            .generate_transform_key(
-                                &group_priv_key.into(),
-                                &user_master_pub_key.into(),
-                                &auth.signing_private_key().into(),
-                            )?
-                            .into(),
-                    )
-                } else {
-                    None
-                };
-                (group_pub_key, encrypted_group_key, transform_key)
-            })
+) -> impl Future<Item = GroupCreateResult, Error = IronOxideErr> + 'a {
+    user_api::user_key_list(auth, members)
+        .and_then(move |member_ids_and_keys| {
+            // this will occur when one of the UserIds in the members list cannot be found
+            if member_ids_and_keys.len() != members.len() {
```

Comment:
Maybe this branch should be its own function? Would this be reusable when adding admins? It certainly would be more testable on its own, IMO.